google-cloud/src/pubsub/subscription.rs

```diff
@@ -64,6 +64,14 @@ impl Subscription {
         }
     }
 
+    /// Returns the unique identifier within its project
+    pub fn id(&self) -> &str {
+        self.name
+            .rfind('/')
+            .map(|i| &self.name[i + 1..])
+            .expect("bad subscription name")
```

Comment:
I feel like the following would be simpler, because it removes the manual sub-slicing operation:
```suggestion
        self.name
            .split("/")
            .last()
            .expect("empty subscription name")
```
Except if this was done to ensure there is at least one slash in `self.name`.  
In which case, I think this is not the place to do this check, because if there wasn't any slashes and the user did not call this method, no errors would have been observed.

Reply:
I was translating from the Go implementation. You made a good point about wrong place for checking existence of slashes! I've pushed a rebased commit with `self.name.rsplit('/').next().unwrap()`, since empty string will just return empty again.